packages/url_launcher/url_launcher_ios/pigeons/messages.dart

```diff
@@ -9,22 +9,55 @@ import 'package:pigeon/pigeon.dart';
   swiftOut: 'ios/Classes/messages.g.swift',
   copyrightHeader: 'pigeons/copyright.txt',
 ))
+
+/// Possible outcomes of launching a URL.
+enum LaunchResult {
+  /// The URL was successfully launched.
+  success,
+
+  /// The URL could not be launched
+  failure,
+
+  /// The URL was not launched because it is not invalid URL
+  invalidUrl,
+
+  /// The URL did not load successfully in the SFSafariViewController.
+  failedToLoad,
+}
+
+class LaunchResultDetails {
+  LaunchResultDetails({
+    required this.result,
+    this.errorMessage,
+    this.errorDetails,
+  });
+
+  /// The result of the launch attempt.
+  final LaunchResult result;
+
+  /// A system-provided error message, if any.
+  final String? errorMessage;
+
+  /// A system-provided error details, if any.
+  final String? errorDetails;
+}
+
 @HostApi()
 abstract class UrlLauncherApi {
   /// Returns true if the URL can definitely be launched.
   @ObjCSelector('canLaunchURL:')
-  bool canLaunchUrl(String url);
+  LaunchResultDetails canLaunchUrl(String url);
```

Comment:
We should have two different enums, since the `failedToLoad` is specific to Safari view controller, so having it be an enum option here and in `launchUrl` is confusing.